From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: S A Muqthyar Ahmed <syed.abdul.muqthyar.ahmed@intel.com>
Date: Tue, 29 Oct 2024 19:33:37 +0530
Subject: autotools/m4: manage prototype changes in strscpy

strscpy has moved to dynamic size calculation and only 2 arguments in
KV6.9. use alternative for older kernels.

Error:
--------------------------------------------------------------------------
error: too few arguments to function _strscpy
--------------------------------------------------------------------------

Reference-id:
e6584c3964f2 string: Allow 2-argument strscpy()

Signed-off-by: S A Muqthyar Ahmed <syed.abdul.muqthyar.ahmed@intel.com>
Signed-off-by: Kanaka Raju Nayana <kanaka.raju.nayana@intel.com>
---
 drivers/gpu/drm/xe/xe_devcoredump.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/xe/xe_devcoredump.c b/drivers/gpu/drm/xe/xe_devcoredump.c
index 6980304..4e191eb 100644
--- a/drivers/gpu/drm/xe/xe_devcoredump.c
+++ b/drivers/gpu/drm/xe/xe_devcoredump.c
@@ -292,8 +292,11 @@ static void devcoredump_snapshot(struct xe_devcoredump *coredump,
 		process_name = q->vm->xef->process_name;
 		ss->pid = q->vm->xef->pid;
 	}
-
+#ifdef BPM_STRSCPY_ARG3_PRESENT
+	strscpy(ss->process_name, process_name, sizeof(ss->process_name));
+#else
 	strscpy(ss->process_name, process_name);
+#endif
 
 	ss->gt = q->gt;
 	INIT_WORK(&ss->work, xe_devcoredump_deferred_snap_work);
-- 
2.25.1

